src/diffusers/utils/testing_utils.py

```diff
@@ -860,7 +860,7 @@ def _is_torch_fp64_available(device):
         return True
 
     except Exception as e:
-        if device.type == "cuda":
```

Comment:
Actually the issue seems to only be with the fp64 check. I think we can just convert the `device` str to device object like we have in the fp16 test. 

```python 
device = torch.device(device)
```

Reply:
You're right - updated!